xbmc/video/dialogs/GUIDialogSubtitleSettings.cpp

```diff
@@ -105,15 +106,30 @@ void CGUIDialogSubtitleSettings::OnSettingChanged(std::shared_ptr<const CSetting
 
 std::string CGUIDialogSubtitleSettings::BrowseForSubtitle()
 {
+  bool hasRar = false;
+  for (const auto& vfsAddon : CServiceBroker::GetVFSAddonCache().GetAddonInstances())
+  {
+    if (vfsAddon->GetProtocols().find("rar") != std::string::npos)
+    {
+      hasRar = true;
+      break;
+    }
+  }
+
   std::string strPath;
   if (URIUtils::IsInRAR(g_application.CurrentFileItem().GetPath()) || URIUtils::IsInZIP(g_application.CurrentFileItem().GetPath()))
     strPath = CURL(g_application.CurrentFileItem().GetPath()).GetHostName();
   else
     strPath = g_application.CurrentFileItem().GetPath();
 
-  std::string strMask = ".utf|.utf8|.utf-8|.sub|.srt|.smi|.rt|.txt|.ssa|.aqt|.jss|.ass|.idx|.rar|.zip";
+  std::string strMask = ".utf|.utf8|.utf-8|.sub|.srt|.smi|.rt|.txt|.ssa|.aqt|.jss|.ass|.idx|.zip";
+
   if (g_application.GetCurrentPlayer() == "VideoPlayer")
-    strMask = ".srt|.rar|.zip|.ifo|.smi|.sub|.idx|.ass|.ssa|.txt";
+    strMask = ".srt|.zip|.ifo|.smi|.sub|.idx|.ass|.ssa|.txt";
+
+  if (hasRar)
+    strMask += "|.rar";
```

Comment:
Wouldn't it be better to add all extensions listed in (all enabled) vfs addon.xml instead of relying on hard-coded values?

Reply:
no. this is a list of subtitle formats.

Reply:
Subtitles could also be in 7z, tar.gz, .tar.bz2, ... which vfs.libarchive lists as supported extensions.
I just would like to avoid hard coding rar into Kodi code, as it isn't a Kodi core feature anymore and adds a special case for an addon.
I can't see any trouble my suggestion could cause, so if you have one in mind please tell me.

Reply:
okay, i get what you mean. but that means i have to hardcode vfs.rar and vfs.libarchive instead.

Reply:
I had the following in mind.
```
for (const auto& vfsAddon : CServiceBroker::GetVFSAddonCache().GetAddonInstances())
{
  const std::string& extensions = vfsAddon->GetExtensions();
  if (!extensions.empty())
    strMask += "|" + extensions;
}
```

Reply:
which would allow selecting sacd isos, and whatever else for subtitles.

Reply:
How about adding `vfsAddon->GetProtocols().find("rar") != std::string::npos || vfsAddon->GetProtocols().find("archive") != std::string::npos` to the if?
With this at least the supported extension wouldn't be hard coded anymore.